Fixed issue with overriding of property mappings.

archfz opened the pull request:
I have fixed a critical issue with the mappings. Property mappings were mixing between child classes extending same parent class. Modified the tests to include this case as well.

This is why it is important in open source development to not close peoples merge request and apply the changes yourself. This was properly handled in the code I wrote, but you modified and changed the behavior. Please this time don't close the MR, if you want things differently add a comment and I will make the changes.

andreas-aeschlimann commented:
Hello, I appreciate your investment in this case. While your commit of the decorator seems to work and it makes sense to me, you did not provide a test case where the old implementation failed.

I ran `npm test` with both your version and the old version and it didn't fail with the test you committed. Could you provide a simple example that makes the old version fail?

archfz commented:
Hello @andreas-aeschlimann . Yes you were right the test was incomplete. I have fixed it here https://github.com/dhlab-basel/json2typescript/pull/86

The issue was that I have added a different mapping name for Cat which overridden the mapping on the parent class Animal, thus the dog could no more accept "name" but was expecting "catName" for the dog name, but the test weren't testing the deserialization of the dog.

andreas-aeschlimann commented:
Thanks for that, this will for sure help pretending issues in the future.

In the meantime, I have added some other smaller request features.
